app/src/main/java/com/ddoczi/tasky/agenda/data/remote/worker/SyncLocalWithRemoteWorker.kt

```diff
@@ -0,0 +1,58 @@
+package com.ddoczi.tasky.agenda.data.remote.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.ddoczi.tasky.agenda.data.local.AgendaDao
+import com.ddoczi.tasky.agenda.data.mapper.toEventEntity
+import com.ddoczi.tasky.agenda.data.mapper.toReminderEntity
+import com.ddoczi.tasky.agenda.data.mapper.toTaskEntity
+import com.ddoczi.tasky.agenda.data.remote.AgendaApi
+import com.ddoczi.tasky.agenda.data.remote.dto.AgendaResponseDto
+import com.ddoczi.tasky.agenda.domain.model.AgendaItem
+import com.ddoczi.tasky.core.util.toLong
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import java.time.LocalDateTime
+
+@HiltWorker
+class SyncLocalWithRemoteWorker @AssistedInject constructor(
+    @Assisted val context: Context,
+    @Assisted val workerParameters: WorkerParameters,
+    private val api: AgendaApi,
+    private val dao: AgendaDao,
+) : CoroutineWorker(context, workerParameters) {
+    override suspend fun doWork(): Result {
+
+        val agenda = try {
+            getAgenda()
+        } catch (e: Exception) {
+            return Result.failure()
+        }
+
+        val items = agenda.events.map { it } + agenda.reminders.map { it } + agenda.tasks.map { it }
+
+        supervisorScope {
+            val jobs = items.map { launch { saveItem(it) } }
+            jobs.forEach { it.join() }
+        }
+        return Result.success()
+    }
+
+    private suspend fun saveItem(item: Any) {
+        when (item) {
+            is AgendaItem.Reminder -> { dao.insertReminder(item.toReminderEntity()) }
+            is AgendaItem.Task -> { dao.insertTask(item.toTaskEntity()) }
+            is AgendaItem.Event -> {
+                supervisorScope {
+                    dao.insertEvent(item.toEventEntity())
+                }
+            }
+        }
+    }
+
+    private suspend fun getAgenda() : AgendaResponseDto = api.getAgenda(LocalDateTime.now().toLong())
```

Comment:
Do you feel like you gain a lot by extracting this simple call into a separate function? This just requires you to jump down in order to understand what this really does.

Also for syncing local with remote, you should use the /fullAgenda endpoint which returns all items. Otherwise you won't see older/newer agenda items if you go offline after a successful sync